batch/batch/worker/worker.py

```diff
@@ -761,7 +997,12 @@ def __init__(
         self.main_volume_mounts = []
         self.output_volume_mounts = []
 
-        io_volume_mount = f'{self.io_host_path()}:/io'
+        io_volume_mount = {
+            'source': self.io_host_path(),
+            'destination': '/io',
+            'type': 'none',
+            'options': ['rbind', 'rw'],
+        }
```

Comment:
Thought: now that we fully control the filesystem, can we rid ourselves of this duplicated /io and /? Can we place the writable layers wherever the io_host_path is?

Reply:
Yes we can and should do that. I'm not sure what can of worms that might open up, but we can put the writeable layer anywhere. Ideally a user's xfs project should just have one root directory.

Reply:
I moved the container's overlay and crun config into the job's scratch directory, which is great because I could then remove the extra call to `xfs_quota` to add the overlay to the xfs project. Here is basically the directory tree for a job:

```
scratch (actually /batch/token)/
├─ io/
├─ input/
│  ├─ rootfs_overlay/
│  │  ├─ upperdir/
│  │  │  ├─ io/ (bind mount)
│  │  ├─ merged/
│  │  ├─ workdir/
│  ├─ volumes/
│  ├─ config/
│  │  ├─ config.json
├─ main/
│  ├─ rootfs_overlay/
│  │  ├─ upperdir/
│  │  │  ├─ io/ (bind mount)
│  │  ├─ merged/
│  │  ├─ workdir/
│  ├─ volumes/
│  ├─ config/
│  │  ├─ config.json
├─ output/
…
```

where `scratch` (like before) is marked with the user's storage quota for that job. Note that this does not contain the `lowerdir` of the overlays, because those are shared, readonly extracted rootfs's that we share across containers/overlays.

Reply:
Perhaps it's not fair to make the user pay for the size of the config, and we could store that elsewhere, but it's a tiny json file…

Reply:
Can we add that directory tree to a dev-doc? This is a great visual.

Reply:
And just to be clear: with this change the entire disk quota should be available at any path, right? Can we add a couple tests to that effect? Try starting a 1 cpu job with 10GB and writing nearly 15GB to `/foo` and `/io/foo`.

Reply:
cc: @jigold

Reply:
Yes. Added a test for that and added a brief dev doc for a job's layout on the filesystem.